src/zarr/core/array_spec.py

```diff
@@ -102,7 +102,9 @@ def __init__(
     ) -> None:
         shape_parsed = parse_shapelike(shape)
         dtype_parsed = np.dtype(dtype)
-        fill_value_parsed = parse_fill_value(fill_value)
+        # No zarr_format available here...
+        # fill_value_parsed = parse_fill_value(fill_value, dtype, zarr_format=2)
+        fill_value_parsed = fill_value
```

Comment:
Not sure what to make of this. The fill value is parsed way up the stack from here or else the dtype we have to inspect is lost. Should I just toss this out?